contracts/pcv/PCVRouter.sol

```diff
@@ -0,0 +1,242 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.13;
+
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+import {IPCVRouter} from "./IPCVRouter.sol";
+import {IPCVSwapper} from "./IPCVSwapper.sol";
+import {CoreRefV2} from "../refs/CoreRefV2.sol";
+import {IPCVDeposit} from "./IPCVDeposit.sol";
+import {VoltRoles} from "../core/VoltRoles.sol";
+import {PCVOracle} from "../oracle/PCVOracle.sol";
+
+/// @title Volt Protocol PCV Router
+/// @notice A contract that allows PCV movements between deposits.
+/// @dev This contract requires the PCV_CONTROLLER role.
+/// @author eswak
+contract PCVRouter is IPCVRouter, CoreRefV2 {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    ///@notice set of whitelisted PCV swappers
+    EnumerableSet.AddressSet private pcvSwappers;
```

Comment:
would there ever be a reason to keep this whitelisted set of swappers in another contract?

Reply:
probably not